Fix(eos_designs): Pull endpoint interface information from combined adapter+port profile settings

PhirePhly opened the pull request:
## Change Summary

endpoint_ports was incorrectly being pulled from the original raw adapter
configuration, which did not allow the endpoint port to be configured in
a port profile and inherited, which would be useful when stamping out a large number of
connected endpoints with the same port configuraiton.

Also makes it much easier to stub out endpoint_ports when they're not
applicable for the user since they can set endpoint_ports: [ 'N/A' ] in
just a parent port profile template.

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Pull endpoint_ports from the merged dataset instead of the original top level dict

## How to test

Moved one server_ports definition up into a profile.

## Checklist

### Repository Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

PhirePhly committed:
Fix(eos_designs): Pull endpoint_ports from combined settings

endpoint_ports was incorrectly being pulled from the original raw adapter
configuration, which did not allow the endpoint port to be configured in
a port profile, which would be useful when stamping out a large number of
connected endpoints with the same port configuraiton.

Also makes it much easier to stub out endpoint_ports when they're not
applicable for theuser since they can set endpoint_ports: [ 'N/A' ] in
just a parent port profile template.

PhirePhly committed:
Test(eos_designs): Support inheriting server_ports/endpoint_ports from port profile

tgodaA commented on ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2:
```diff
@@ -12,7 +12,7 @@ ethernet_interfaces:
+{%             set endpoint_ports = adapter_settings.endpoint_ports | arista.avd.default(adapter_settings.server_ports) %}
 {%             set peer_type = connected_endpoints_keys[endpoint_key].type %}
 {%             set peer = connected_endpoint %}
 {%             for adapter_switch in adapter.switches %}
```
@ClausHolbechArista, I think we can modify this parameter, ```adapter.switches``` to ```adapter_settings.switches```. This gives the user a lot of flexibility. #e.g. 20 end_points connected to the same group of switches. So if we add this to profile settings, the user can use the profile settings. Anyway we have the local settings to override the default ones.

PhirePhly replied:
I added a third commit to fix the switch side as well, in case someone wanted to have port profile strides in that dimension instead of in the server interface dimension like I did.

PhirePhly committed:
Fix(eos_designs): Pull connected endpoints switch/switch port from adapter_settings

This allows the switch and switch port to also be specified in port profiles,
such that if a group of connected endpoints have a stride across identical ports
on switches, or are all attached to the same switch, this fact can be moved up
into the port profile and doesn't need to be specified for every endpoint.

ClausHolbechArista reviewed (commented):
`switch_ports`is also referenced in the `port-channel-interfaces.j2`, to calculate the `channel_group_id`. This should also be updated to take it from adapter_settings.

PhirePhly committed:
Fix(eos_designs): Pull connected endpoints port channel switch from combined adapter+port profile

Adds support for defining the switch that port channels are attached to in
port profiles such that multiple port channels can be stamped out with the
same upstream switches.

The ESI and LACP ID is still pulled from the base adapter configuration, since
there is no meaningful way to reuse ESIs or LACP IDs between port channels.

tgodaA commented:
But ```switch_ports``` are unique per adapter. So this can be referenced from ```adapter```.

PhirePhly commented:
It depends on what dimension you're using in the template.

1. Every endpoint uses the same NIC: define `endpoint_ports` in the port profile. i.e. Every server uses their `enp130s2` NIC
2. Every endpoint is connected to the same TOR switch: define `switches` in the port profile. i.e. Every load balancer is connected to the service leaf.
3. Every endpoint is connected to the same port on their TOR switches: define `switch_ports` in the profile. i.e. Every provisioning server is connected to Eth1 on the TORs. 

I only care about scenario 1, but I can believe someone hitting scenario 2. Scenario 3 seems possible but the least likely. 

I'm really just trying to avoid having to define `endpoint_ports: ['N/A']` 140 times for inter-AS peering ports.

ClausHolbechArista reviewed (approved):
LGTM

tgodaA reviewed (approved):
LGTM

carlbuchmann reviewed (approved):
LGTM!
